scopes/dependencies/modules/packages-excluder/generate-node-modules-pattern.spec.ts

```diff
@@ -0,0 +1,52 @@
+import { generateNodeModulesPattern } from './generate-node-modules-pattern';
+
+describe('generateNodeModulesPattern()', () => {
+  it('should return an empty regex when the received array is empty', () => {
+    expect(generateNodeModulesPattern({ packages: [] })).toEqual('node_modules/(?!()/)');
+  });
+  it('should work with more than one packages', () => {
+    const packagesToTransform = ['react', '@myorg', 'testing-library__dom'];
+    const pattern = generateNodeModulesPattern({ packages: packagesToTransform });
+    const regex = new RegExp(pattern);
+    expect(pattern).toEqual(
+      'node_modules/(?!(react|.pnpm/registry.npmjs.org/react.*|.pnpm/react.*|@myorg|.pnpm/registry.npmjs.org/@myorg.*|.pnpm/@myorg.*|testing-library__dom|.pnpm/registry.npmjs.org/testing-library__dom.*|.pnpm/testing-library__dom.*)/)'
+    );
+    expect(regex.test('node_modules/react')).toBeTruthy();
```

Comment:
the regex only check internal paths not the dir. so for example instead of `expect(regex.test('node_modules/react')).toBeTruthy();` you should have `expect(regex.test('node_modules/react/some-path')).toBeTruthy();` see the some-path also the value should be false not true. if it's true it means it's not working :slightly_smiling_face: (you got true because you didn't put an internal path)
That's relevant for all the tests you did

Reply:
ok, thank you very much for this review.

Reply:
done

Reply:
done